libraries/botbuilder-core/botbuilder/core/bot_framework_adapter.py

```diff
@@ -965,3 +968,31 @@ def check_emulating_oauth_cards(self, context: TurnContext):
             )
         ):
             self._is_emulating_oauth_cards = True
+
+    async def _ensure_channel_connector_client_is_created(
+        self, service_url: str, claims_identity: ClaimsIdentity
+    ):
+        # Ensure we have a default ConnectorClient and MSAppCredentials instance for the audience.
+        audience = claims_identity.claims.get(AuthenticationConstants.AUDIENCE_CLAIM)
+
+        if (
+            not audience
+            or AuthenticationConstants.TO_BOT_FROM_CHANNEL_TOKEN_ISSUER != audience
+        ):
+            # We create a default connector for audiences that are not coming from
+            # the default https://api.botframework.com audience.
+            # We create a default claim that contains only the desired audience.
+            default_connector_claims = {
+                AuthenticationConstants.AUDIENCE_CLAIM: audience
+            }
+            connector_claims_identity = ClaimsIdentity(
+                claims=default_connector_claims, is_authenticated=True
+            )
+
+            await self.create_connector_client(service_url, connector_claims_identity)
+
+        if SkillValidation.is_skill_claim(claims_identity.claims):
+            # Add the channel service URL to the trusted services list so we can send messages back.
+            # the service URL for skills is trusted because it is applied by the
+            # SkillHandler based on the original request received by the root bot
+            MicrosoftAppCredentials.trust_service_url(service_url)
```

Comment:
This is fine as it mirrors the dotnet impl, but last time we spoke in person you were considering following the [JS implementation (js-#1586)](https://github.com/microsoft/botbuilder-js/pull/1586) which places this logic in the SkillHandler class. @gabog indicated he was going to mirror the JS implementation as well regarding this check for R8.

I was curious to know if there have been any updates or changes re: dotnet moving this check to the SkillHandler class.

Reply:
Oh ok, I was planning to move it at the same time C# does (didn't know that was happening in R8). I do think JS way is better, is just that because I was aligning to C# I wanted to be consistent with the code parity

Reply:
No problem! I just wanted to know if I needed to change JS 😊